modules/@apostrophecms/piece-type/index.js

```diff
@@ -507,6 +515,18 @@ module.exports = {
                 return operations.includes(operation);
               }) || [];
           }
+        },
+        composeUtilityOperations() {
+          self.utilityOperations = Object.entries(self.utilityOperations || {})
+            .reduce((acc, [ action, properties ]) => {
```

Comment:
You use `reduce` well, but I don't know if I'll ever understand its pattern easily.

Reply:
Basically here I just change an object to an array, the last parameter is an empty array, that means that At my first iteration the accumulator is an empty array, and I simply return this array with the old value with a spread `...acc`, and I add a new object in this array. The `Object.entries` allows me to get the key (action) and its object (properties) to put it together in an object.
It's purpose is to iterate through an array and to reduce this array in something else. So this is a perfect way to change an object in array, or in string or in anything that is not an object.
Like map and filter it's functional oriented so it's always return a new element, that's why it fits perfectly with the spread pattern.

Reply:
I do understand what is happening, but I don't use it often so it usually takes me a few minutes to figure it out again.

Reply:
In this case you're right it's not necessary, a simple map is enough and more readable, changed

Reply:
Yes I understand in this case I definitely over used it.

Reply:
lol I wasn't trying to suggest you change it, but this works as well.